Dockerfile

```diff
@@ -0,0 +1,55 @@
+# First, get the elixir dependencies within an elixir container
+FROM hexpm/elixir:1.11.3-erlang-23.2.5-alpine-3.13.1 AS elixir-builder
+
+ENV LANG="C.UTF-8" MIX_ENV=prod
+
+WORKDIR /root
+ADD . .
+
+# Install hex, rebar, and deps
+RUN mix local.hex --force && \
+  mix local.rebar --force && \
+  mix do deps.get --only prod
+
+# Next, build the frontend assets within a node.js container
+FROM node:14.15-alpine as assets-builder
+
+WORKDIR /root
+ADD . .
+
+# Copy in elixir deps required to build node modules for phoenix
+COPY --from=elixir-builder /root/deps ./deps
+
+RUN npm --prefix assets ci
+RUN npm --prefix assets run deploy
+
+# Now, build the application back in the elixir container
+FROM elixir-builder as app-builder
+
+ENV LANG="C.UTF-8" MIX_ENV=prod
+
+WORKDIR /root
+
+# Add frontend assets compiled in node container, required by phx.digest
+COPY --from=assets-builder /root/priv/static ./priv/static
+
+RUN mix do compile --force, phx.digest, release
+
+# Finally, use a Debian container for the runtime environment
+FROM alpine:latest
+
+RUN apk add --update libssl1.1 ncurses-libs \
+  && rm -rf /var/cache/apk
+
+WORKDIR /root
+EXPOSE 4000
+ENV MIX_ENV=prod TERM=xterm LANG="C.UTF-8" PORT=4000
+
+# Add frontend assets with manifests from app-builder container
+COPY --from=app-builder /root/priv/static ./priv/static
+
+# Add application artifact compiled in app-builder container
+COPY --from=app-builder /root/_build/prod/rel/document_viewer .
+
+# Run the application
+CMD ["bin/document_viewer", "start"]
```

Comment:
Some other steps here in the last stage (example from [tablespoon](https://github.com/mbta/tablespoon/blob/HEAD/Dockerfile#L26-L53)):
- create a separate user instead of running as root
- add `dumb-init` package and create it as the `ENTRYPOINT` so that signals (such as SIGTERM to shut down the application) are properly handled
- add a `HEALTHCHECK` that the node is running (I don't recall whether `rpc` is part of the native release or Distillery though)

Reply:
`dumb-init` is listed in the [Application Checklist](https://github.com/mbta/wiki/blob/HEAD/devops/application-guides/new-application/2-early-development.md) but the others are note: I'll put together a PR with those updates.

Reply:
@paulswartz Thanks, added those. One final question is, should I be able to run the container locally? The build is working just fine, but when I run it nothing shows up in the logs. If I log into a shell on the container I can see beam running, but get a `connection refused` trying to hit port 4000. Is that typical?

Reply:
You need to map the port and provide the environment when you're running locally: something like `docker build -t document_viewer; docker run -it --rm --env-file .env -p 4000:4000 document_viewer` should work.